### ISSUE TITLE
Support for Loss Matrix (Cost Sensitive Classification)

### ISSUE THREAD

**Issue by RoyiAvital** [adamingas/ordinalgbt#10]
Have you thought on the following case, Cost Sensitive Classification:

$$
\boldsymbol{L} = \begin{bmatrix} l \left( y = 1, \hat{y} = 1  \right) & l \left( y = 1, \hat{y} = 2  \right) & \dots & l \left( y = 1, \hat{y} = K  \right) \\\\ l \left( y = 2, \hat{y} = 1  \right) & l \left( y = 2, \hat{y} = 2  \right) & \dots & l \left( y = 2, \hat{y} = K  \right) \\\\ \vdots & \vdots & \vdots & \vdots \\\\ l \left( y = K, \hat{y} = 1  \right) & l \left( y = K, \hat{y} = 2  \right) & \dots & l \left( y = K, \hat{y} = K  \right) \end{bmatrix}
$$

Is there a way to implement is where the Gradient and Hessian are easy to calculate for `LightGBM`?

**Comment by adamingas** [comment]
Hello, sorry for not getting back to you, I don't receive notifications for issues and I need to fix that. Will take a look over the weekend to see if something like this could be implemented.

**Comment by RoyiAvital** [comment]
@adamingas , On `Settings -> Notifications` make sure you have `GitHub` in the list:

![image](https://github.com/user-attachments/assets/542cab4b-6d24-4366-a394-2d2b67574012)

It would be great to have a package with Cost Sensitive classification.  
Even just as a demo wrapper for `XGBoost` / `LightGBM`.
